src/types.rs

```diff
@@ -741,6 +741,7 @@ impl SourceMap {
         self.sources_content
             .get(idx as usize)
             .and_then(Option::as_ref)
+            .and_then(|v| unsafe { std::mem::transmute(v) })
```

Comment:
I meet the error when run lint. I'm also not sure why it happened.

```
error: lifetime may not live long enough
   --> src/types.rs:741:9
    |
740 |       pub fn get_source_view(&self, idx: u32) -> Option<&SourceView<'_>> {
    |                              - let's call the lifetime of this reference `'1`
741 | /         self.sources_content
742 | |             .get(idx as usize)
743 | |             .and_then(Option::as_ref)
    | |_____________________________________^ returning this value requires that `'1` must outlive `'static`
    |
```

Reply:
The changes of `DecodedMap` is also similar error.

Reply:
I tried this locally and I believe I figured this out.

There is also a NOTE attached to that error:

```
    = note: requirement occurs because of the type `SourceView<'_>`, which makes the generic argument `'_` invariant
    = note: the struct `SourceView<'a>` is invariant over the parameter `'a`
```

I believe that with the fact that `SourceMap` defines things as `pub(crate) sources_content: Vec<Option<SourceView<'static>>>,` somehow means that the `SourceView` needs to be `'static`. Or at least, the `&str` inside it.

Long story short, I changed the lifetime in `lines` like this: `lines: Mutex<Vec<&'static str>>,`, and now things are working without changes to `types.rs`.

It would be nice to also add a `SAFETY` note to the (I believe) one line of `unsafe` that remains explaining why we believe this to be a safe use.

Reply:
OK. Thanks a lot for your help.